compiler/rustc_mir_build/src/thir/pattern/deconstruct_pat.rs

```diff
@@ -697,6 +697,7 @@ impl<'tcx> Constructor<'tcx> {
     /// Returns whether `self` is covered by `other`, i.e. whether `self` is a subset of `other`.
     /// For the simple cases, this is simply checking for equality. For the "grouped" constructors,
     /// this checks for inclusion.
+    #[inline]
     pub(super) fn is_covered_by<'p>(&self, pcx: PatCtxt<'_, 'p, 'tcx>, other: &Self) -> bool {
```

Comment:
This is an awfully big function to inline :eyes:

Reply:
Ah, I did not take this into consideration. Given that it's one of the only functions that's repeatedly called between `usefulness` and `deconstruct_pat`, and inlining it precisely restored perf, I gathered that it used to be automatically inlined before I split the file.

Reply:
Also it's got a single call site

Reply:
Ah ok, if there's only one call site that makes sense.

Reply:
Maybe add that as a comment?

Reply:
Done